crates/cxx-qt-gen/src/parser/signals.rs

```diff
@@ -42,12 +42,13 @@ impl ParsedSignal {
         let fields = MethodFields::parse(method, auto_case)?;
         let attrs = require_attributes(&fields.method.attrs, &Self::ALLOWED_ATTRS)?;
 
-        if !fields.mutable {
-            return Err(Error::new(
-                fields.method.span(),
-                "signals must be mutable, use Pin<&mut T> instead of T for the self type",
-            ));
-        }
+        // TODO: add proper checks
```

Comment:
right, as @LeonMatthesKDAB suggested, we probably only want const signals support inside `unsafe extern "C++Qt"` blocks not `unsafe extern "RustQt"`, so we need to determine that info somehow here :thinking: